app/transport/rest/user.go

```diff
@@ -60,3 +68,40 @@ func (uh UserHandler) CreateUser() http.Handler {
 		uh.logger.Debugw("User successfully created", "ID", id)
 	})
 }
+
+// GetUserByID will handle user search.
+func (uh UserHandler) GetUserByID() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		id := mux.Vars(req)["id"]
+
+		if err := validateUUID(id); err != nil {
+			uh.logger.Warnw("Invalid request", "ID", id)
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+
+			return
+		}
+
+		entUser, err := uh.usecase.GetUser(id)
+		if err != nil {
+			if errors.Is(err, globs.ErrNotFound) {
+				uh.logger.Debugw("No user found.", "ID", id)
+				WriteJSONResponse(w, http.StatusNotFound, Response{Message: MsgUserNotFound}, uh.logger)
```

Comment:
`MsgUserNotFound` - usually the message describes the error in general like `Bad Request` or `Resource Not Found`. If you say specifically that the User was not found, I'd say it's more often written in details.

Reply:
fixed. I think so...

Reply:
And you also actually don't need the `MsgUserNotFound` since messages are usually more generic. Some `MsgNotFound` should be fine.

Reply:
fixed